### PR TITLE
Fix #6046, Password: (accessiblility) cannot access toggle button using keyboard

### LINKED ISSUE #6046: Password: (accessiblility) cannot access toggle button using keyboard
### Describe the bug

focus cannot be moved to Password's toggle button, then cannot toggle using keyboard

### Reproducer

_No response_

### PrimeReact version

10.5.1

### React version

17.x

### Language

TypeScript

### Build / Runtime

Create React App (CRA)

### Browser(s)

_No response_

### Steps to reproduce the behavior

1. go to Password toggle show case : https://primereact.org/password/#togglemask
2. use tab to set focus on password field
3. hit tab several times, focus is never set on toggle password button

### Expected behavior

Toggle button should be able to be activated using keyboard
Focus should also stay on toggle button when password state is toggled.

### PR DESCRIPTION
**Pull request by akshayaqburst**
### Defect Fixes
fix #6046, Password: (accessiblility) cannot access toggle button using keyboard

### CODE REVIEW

File: components/lib/password/Password.js [hunk 1]
```diff
@@ -272,16 +272,21 @@ export const Password = React.memo(
             ZIndexUtils.clear(overlayRef.current);
         });
 
+        const onEyeIconKeyPress = (e) => {
```
Comment (melloware) [inline components/lib/password/Password.js:275]:
```suggestion
        const onToggleMaskKeyDown= (e) => {
```

File: components/lib/password/Password.js [hunk 2]
```diff
@@ -272,16 +272,21 @@ export const Password = React.memo(
...
             const hideIconProps = mergeProps(
                 {
                     key: 'hideIcon',
                     role: 'button',
-                    tabIndex: props.tabIndex,
+                    tabIndex: props.tabIndex || '0',
                     className: cx('hideIcon'),
-                    onClick: toggleMask
+                    onClick: toggleMask,
+                    onKeyDown: onEyeIconKeyPress
```
Comment (melloware) [inline components/lib/password/Password.js:289]:
```suggestion
                    onKeyDown: onToggleMaskKeyDown
```

File: components/lib/password/Password.js [hunk 3]
```diff
@@ -290,9 +295,10 @@ export const Password = React.memo(
                 {
                     key: 'showIcon',
                     role: 'button',
-                    tabIndex: props.tabIndex,
+                    tabIndex: props.tabIndex || '0',
                     className: cx('showIcon'),
-                    onClick: toggleMask
+                    onClick: toggleMask,
+                    onKeyDown: onEyeIconKeyPress
```
Comment (melloware) [inline components/lib/password/Password.js:301]:
```suggestion
                    onKeyDown: onToggleMaskKeyDown
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
Also we need to add to both buttons...

`'aria-label': ariaLabel('toggleMaskLabel') || "Toggle Mask"`

**Comment by akshayaqburst** [comment]
@melloware resolved comments.

File: components/lib/password/Password.js [hunk 4]
```diff
@@ -272,16 +272,22 @@ export const Password = React.memo(
             ZIndexUtils.clear(overlayRef.current);
         });
 
+        const onToggleMaskKeyDown = (e) => {
+            return e.key === 'Enter' ? toggleMask() : '';
```
Comment (melloware) [inline components/lib/password/Password.js:276]: Do you need the `return` here?
```suggestion
            e.key === 'Enter' && toggleMask();
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/password/Password.js (hunk 4 above)
Comment (melloware) [inline components/lib/password/Password.js:276]:
```suggestion
            if (event.key === 'Enter' || event.key === 'Space') {
                toggleMask();
                event.preventDefault();
            }
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by melloware** [comment]
Run `npm run format` and it should be good!

**Comment by akshayaqburst** [comment]
@melloware Good to review

**Review by melloware — APPROVED** [review]

**Comment by melloware** [comment]
Thanks this is great!

**Comment by melloware** [comment]
@akshayaqburst I just read this article https://medium.com/kiipco/password-creation-3-ways-to-make-it-accessible-bc8f2b53b7ee#1751

I think we should chnage two more properties to the icons. Change `role="button"` to `"role="switch"` and add `aria-checked`
 
```
     role: "switch",
     "aria-checked": "false"
```

`aria-checked: "true"` true on the `show` button and false on the `hide` button.

File: components/lib/password/Password.js [hunk 5]
```diff
@@ -272,16 +272,25 @@ export const Password = React.memo(
...
                 {
                     key: 'hideIcon',
                     role: 'button',
-                    tabIndex: props.tabIndex,
+                    tabIndex: props.tabIndex || '0',
                     className: cx('hideIcon'),
-                    onClick: toggleMask
+                    onClick: toggleMask,
+                    onKeyDown: onToggleMaskKeyDown,
+                    'aria-label': ariaLabel('toggleMaskLabel') || 'Toggle Password Mask'
```
Comment (melloware) [inline components/lib/password/Password.js:293]:
```suggestion
                    'aria-label': ariaLabel('togglePasswordLabel') || 'Toggle Password Mask'
```

**Review by melloware — CHANGES_REQUESTED** [review]

File: components/lib/password/Password.js [hunk 6]
```diff
@@ -272,16 +272,25 @@ export const Password = React.memo(
...
+            }
+        };
+
         const createIcon = () => {
             let icon;
 
             const hideIconProps = mergeProps(
                 {
                     key: 'hideIcon',
                     role: 'button',
```
Comment (melloware) [inline components/lib/password/Password.js:288]:
```suggestion
                    role: 'switch',
```

File: components/lib/password/Password.js [hunk 7]
```diff
@@ -290,9 +299,11 @@ export const Password = React.memo(
                 {
                     key: 'showIcon',
                     role: 'button',
```
Comment (melloware) [inline components/lib/password/Password.js:301]:
```suggestion
                    role: 'switch',
```

File: components/lib/password/Password.js [hunk 8]
```diff
@@ -290,9 +299,11 @@ export const Password = React.memo(
...
-                    tabIndex: props.tabIndex,
+                    tabIndex: props.tabIndex || '0',
                     className: cx('showIcon'),
-                    onClick: toggleMask
+                    onClick: toggleMask,
+                    onKeyDown: onToggleMaskKeyDown,
+                    'aria-label': ariaLabel('toggleMaskLabel') || 'Toggle Password Mask'
```
Comment (melloware) [inline components/lib/password/Password.js:306]:
```suggestion
                    'aria-label': ariaLabel('togglePasswordLabel') || 'Toggle Password Mask',
                    'aria-checked': 'true'
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by akshayaqburst** [comment]
@melloware Comments resolved. Thanks,

**Comment by melloware** [comment]
OK later today I might have one more change for you i am going to update the labels in PrimeLocale and have you change the aria label once last time.  I think I am going to go with 2 labels.

File: components/lib/password/Password.js [hunk 9]
```diff
@@ -272,27 +272,40 @@ export const Password = React.memo(
...
                     key: 'hideIcon',
-                    role: 'button',
-                    tabIndex: props.tabIndex,
+                    role: 'switch',
+                    tabIndex: props.tabIndex || '0',
                     className: cx('hideIcon'),
-                    onClick: toggleMask
+                    onClick: toggleMask,
+                    onKeyDown: onToggleMaskKeyDown,
+                    'aria-label': ariaLabel('togglePasswordLabel') || 'Toggle Password Mask',
```
Comment (melloware) [inline components/lib/password/Password.js:293]:
```suggestion
                    'aria-label': ariaLabel('passwordHide') || 'Hide Password',
```

File: components/lib/password/Password.js [hunk 10]
```diff
@@ -272,27 +272,40 @@ export const Password = React.memo(
...
                     key: 'showIcon',
-                    role: 'button',
-                    tabIndex: props.tabIndex,
+                    role: 'switch',
+                    tabIndex: props.tabIndex || '0',
                     className: cx('showIcon'),
-                    onClick: toggleMask
+                    onClick: toggleMask,
+                    onKeyDown: onToggleMaskKeyDown,
+                    'aria-label': ariaLabel('togglePasswordLabel') || 'Toggle Password Mask',
```
Comment (melloware) [inline components/lib/password/Password.js:307]:
```suggestion
                    'aria-label': ariaLabel('passwordShow') || 'Show Password',
```

**Review by melloware — CHANGES_REQUESTED** [review]

**Comment by akshayaqburst** [comment]
@melloware Good to review. Thanks,

**Review by melloware — APPROVED** [review]

**Review by nitrogenous — APPROVED** [review]

**Comment by nitrogenous** [comment]
Perfect!

**Pull request merged**